tests/test_instance_billing_type_check.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2024 SUSE Software Solutions Germany GmbH. All rights reserved.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+from unittest.mock import patch, Mock
+from requests import exceptions
+from instance_billing_flavor_check import utils
+
+
+@patch('instance_billing_flavor_check.utils.requests.get')
+def test_make_request_ipv4(mock_request_get):
+    """Test make request with IPv4 without issues."""
+    response = Mock()
+    response.status_code = 200
+    response.json.return_value = {'flavor': 'amazing flavor'}
+    mock_request_get.return_value = response
+    assert utils.make_request('1.1.1.1', 'foo', 'bar') == 'amazing flavor'
+    mock_request_get.assert_called_once_with(
+        'https://1.1.1.1/api/instance/check',
+        timeout=2,
+        verify=False,
+        params={'metadata': 'foo', 'identifier': 'bar'}
+    )
+
+
+@patch('instance_billing_flavor_check.utils.requests.get')
+def test_make_request_ipv6(mock_request_get):
+    """Test make request with IPv6 without issues."""
+    response = Mock()
+    response.status_code = 200
+    response.json.return_value = {'flavor': 'supa flavor'}
+    mock_request_get.return_value = response
+    assert utils.make_request('fc00::1', 'foo', 'bar') == 'supa flavor'
+    mock_request_get.assert_called_once_with(
+        'https://[fc00::1]/api/instance/check',
```

Comment:
This is unique local address but I'd still go for the documentation prefix, e.g. `2001:DB8::1` address. Same applies for further checks. See [RFC3849](https://datatracker.ietf.org/doc/html/rfc3849).